File: leetcode102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        results = []
        levelNodes = [root]
        while root and levelNodes:
            currentNodes = []
            nextLevelNodes = []
            for node in levelNodes:
                currentNodes.append(node.val)
                if node.left:
                    nextLevelNodes.append(node.left)
                if node.right:
                    nextLevelNodes.append(node.right)
            results.append(currentNodes)
            levelNodes = nextLevelNodes
        return results

File: javaStuff/leetcode102.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();
        if(root == null)return result;
        Queue<TreeNode> layer = new LinkedList<>();
        layer.add(root);
        while(layer.size() > 0){
            int counter = layer.size();
            List<Integer> layerContents = new ArrayList<>();
            for(int i = 0; i < counter; i++){
                TreeNode currentNode = layer.remove();
                layerContents.add(currentNode.val);
                if(currentNode.left != null) layer.add(currentNode.left);
                if(currentNode.right != null) layer.add(currentNode.right);
            }
            result.add(layerContents);
        }
        return result;
    }
}